wsd/DocumentBroker.cpp

```diff
@@ -4423,99 +4433,131 @@ bool DocumentBroker::handleInput(const std::shared_ptr<Message>& message)
-                if (firstLine.size() < message->size())
-                    COOLWSD::writeTraceEventRecording(message->data().data() + firstLine.size() + 1,
-                                                      message->size() - firstLine.size() - 1);
+                LOG_CHECK_RET(message->tokens().size() == 1, false);
+                if (COOLWSD::TraceEventFile != NULL && TraceEvent::isRecordingOn())
+                {
+                    const auto& firstLine = message->firstLine();
+                    if (firstLine.size() < message->size())
+                        COOLWSD::writeTraceEventRecording(message->data().data() + firstLine.size() + 1,
+                                                          message->size() - firstLine.size() - 1);
+                }
+                break;
             }
-        }
-        else if (message->firstTokenMatches("forcedtraceevent:"))
-        {
-            LOG_CHECK_RET(message->tokens().size() == 1, false);
-            if (COOLWSD::TraceEventFile != NULL)
+            case KitMessage::ForcedTraceEvent:
             {
-                const auto& firstLine = message->firstLine();
-                if (firstLine.size() < message->size())
-                    COOLWSD::writeTraceEventRecording(message->data().data() + firstLine.size() + 1,
-                                                      message->size() - firstLine.size() - 1);
+                LOG_CHECK_RET(message->tokens().size() == 1, false);
+                if (COOLWSD::TraceEventFile != NULL)
+                {
+                    const auto& firstLine = message->firstLine();
+                    if (firstLine.size() < message->size())
+                        COOLWSD::writeTraceEventRecording(message->data().data() + firstLine.size() + 1,
+                                                          message->size() - firstLine.size() - 1);
+                }
+                break;
+            }
+            case KitMessage::MemoryTrimmed:
+                clearCaches();
+                break;
+            case KitMessage::UnitResult:
+                #if ENABLE_DEBUG
+                UnitWSD::get().processUnitResult(message->tokens());
+                #endif
+                break;
+
+            case KitMessage::Unknown:
+            default:
+            {
+                LOG_ERR("Unexpected message: [" << message->abbr() << ']');
+                return false;
             }
-        }
-        else if (message->firstTokenMatches("memorytrimmed:"))
-        {
-            clearCaches();
-        }
-#if ENABLE_DEBUG
-        else if (message->firstTokenMatches("unitresult:"))
-        {
-            UnitWSD::get().processUnitResult(message->tokens());
-        }
-#endif
-        else
-        {
-            LOG_ERR("Unexpected message: [" << message->abbr() << ']');
-            return false;
         }
     }
-
-    return true;
+    return true; // Final return statement for DocumentBroker::handleInput
```

Comment:
Please maintain a blank line between the closing brace and the next statement for readability.